servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1942,9 +1942,37 @@ public final Publisher<T> concat(Single<? extends T> next) {
      * completion of {@code next} {@link Completable}.
      *
      * @see <a href="https://reactivex.io/documentation/operators/concat.html">ReactiveX concat operator.</a>
+     * @see #concat(Completable, boolean)
      */
     public final Publisher<T> concat(Completable next) {
-        return new PublisherConcatWithCompletable<>(this, next);
+        return concat(next, false);
+    }
+
+    /**
+     * Listens for completion of {@code next} {@link Completable} after {@code this} {@link Publisher} terminates
+     * successfully. Any error from {@code this} {@link Publisher} and {@code next} {@link Completable} is forwarded to
+     * the returned {@link Publisher}.
+     * <p>
+     * This method provides a means to sequence the execution of two asynchronous sources and in sequential programming
+     * is similar to:
+     * <pre>{@code
+     *     List<T> results = resultOfThisPublisher();
+     *     resultOfCompletable(next);
+     *     return results;
+     * }</pre>
+     *
+     * @param next {@link Completable} to wait for completion after {@code this} {@link Publisher} terminates
+     * successfully.
+     * @param propagateCancel if {@code true} then {@code next} will be subscribed to and cancelled if this
+     * {@link Publisher} is cancelled or terminates with {@link Subscriber#onError(Throwable)}. if {@code false} this
+     * method is equivalent to {@link #concat(Completable)}.
+     * @return A {@link Publisher} that emits all items from this {@link Publisher} and then awaits successful
+     * completion of {@code next} {@link Completable}.
+     * @see <a href="https://reactivex.io/documentation/operators/concat.html">ReactiveX concat operator.</a>
+     * @see #concat(Completable)
+     */
+    public final Publisher<T> concat(Completable next, boolean propagateCancel) {
```

Comment:
I'm still unsure about adding an overload vs using a new name `concatPropagateCancel`. I'm not concerned of people moving between sources. Chance someone will move from `Publisher.concat(Completable, true)` to `Single.concat(Publisher, true)` is low. More worried about people memorizing meaning of the 2nd boolean in one case and being surprised the meaning have changed for another use-case. I would be less worried if the order of booleans will be the same, but we can not change it for `Single.concat(Publisher)`.

Reply:
discussed offline. will go with `concatPropagateCancel` for now